### PR TITLE
LPS-145893 Migrate FDS displays in `remote-app-web`

### PR DESCRIPTION
**Pull request by markocikos**
@izaera @javierdearcos I'm not sure who is responsible for `remote-app` nowadays, I'm pinging two of you as I see a lot of your code in the module :slightly_smiling_face: 

### References

- [LPS-145893 Migrate FDS displays in `remote-app-web`](https://issues.liferay.com/browse/LPS-145893)
- [Strategy to rebuild Data Set Display as a DXP app](https://liferay.atlassian.net/wiki/spaces/ENGFRONTENDINFRA/pages/1912931035/LPS-140691+-+Create+a+strategy+to+rebuild+Data+Set+Display+as+a+DXP+app)

### What is the goal?

This is a part of the FDS migration:
- [x]  `search-experiences`
- [x] `objects`
- [ ] `commerce`
- [ ] `remote-app`

In this PR, we are:
- migrating `<clay:data-set-display>` to `<frontend-data-set:classic-display>`
- changing backend API from `frontend-taglib-clay` to API from `frontend-data-set` app. There is **no change in logic**, just a switch of tag and API classes, along with rename of variables to match the new API.
- standardizing pattern of handling FDS objects, based on https://github.com/brianchandotcom/liferay-portal/pull/111833#issuecomment-1009551559. This includes:
  - files containing FDS IDs/names are now named `*FDSNames.java`
  - FDS name constant is camel-case and doesn't have `_DATA_SET_DISPLAY` suffix
  - FDS API objects are in `frontend.data.set` package
  - helper model object `RemoteAppClayDataSetEntry` now contains the `FDS` keyword

### What does it look like?

There are no changes in UI. FDS display should look and behave the same as DSD.

### How to replicate

Open Remote Apps page. It contains FDS display with the list of remote apps.

### CODE REVIEW

**Comment by izaera** [comment]
> I'm not sure who is responsible for remote-app nowadays

Nobody really owns it. It's a cooperative effort :sweat_smile:

**Review by izaera — APPROVED** [review]

File: modules/apps/remote-app/remote-app-web/src/main/java/com/liferay/remote/app/web/internal/constants/RemoteAppAdminFDSNames.java [hunk 1]
```diff
@@ -17,9 +17,8 @@
 /**
  * @author Bruno Basto
  */
-public class RemoteAppAdminConstants {
+public class RemoteAppAdminFDSNames {
```
Comment (javierdearcos) [inline modules/apps/remote-app/remote-app-web/src/main/java/com/liferay/remote/app/web/internal/constants/RemoteAppAdminFDSNames.java:20]: Not sure about this rename. Is it a current pattern?

Anyway let's try with it

**Review by javierdearcos — APPROVED** [review]
Thanks for taking care of this! 🎉

**Comment by javierdearcos** [comment]
I was sure this was already ported because the other day I solved the StatusRenderer bug touching classes in the new Frontend Dataset Display module... Now, this is a mystery to me...

I guess that the Renderers are registered globally and it applied to the Remote Apps although the module was using the old implementation? 🤔

**Comment by markocikos** [comment]
> I was sure this was already ported because the other day I solved the StatusRenderer bug touching classes in the new Frontend Dataset Display module... Now, this is a mystery to me...
> 
> I guess that the Renderers are registered globally and it applied to the Remote Apps although the module was using the old implementation? thinking

@javierdearcos We migrated all frontend parts months ago, by swapping implementation of `<clay:data-set-display>`. Instead of referencing local JS code, we have [react:component](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set/page.jsp#L93-L96) that uses component in [frontend-data-set-web](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set/js/DataSetTag.js#L15). Conversely, new `<frontend-data-set>` tags used to be implemented in the same way, until we recently changed it to [leverage `ReactRenderer`](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-taglib/src/main/java/com/liferay/frontend/data/set/taglib/servlet/taglib/BaseDisplayTag.java#L159-L167).

Unfortunately, a similar trick cannon be done with tags themselves and the API objects. As they are used in code of product team modules, they need to be migrated the 'old fashioned way'.

**Comment by markocikos** [comment]
ci:forward

File: modules/apps/remote-app/remote-app-web/src/main/java/com/liferay/remote/app/web/internal/constants/RemoteAppAdminFDSNames.java (hunk 1 above)
Comment (markocikos) [reply modules/apps/remote-app/remote-app-web/src/main/java/com/liferay/remote/app/web/internal/constants/RemoteAppAdminFDSNames.java:20]: Yeah, that's the naming pattern Brian requested in https://github.com/brianchandotcom/liferay-portal/pull/111833#issuecomment-1009551559

**Pull request closed without merging**